### PR TITLE
Feature/user post db

### PR DESCRIPTION
**Pull request by muneebazam**

### CODE REVIEW

File: app/src/main/java/ch/epfl/favo/user/UserUtil.java [hunk 1]
```diff
@@ -27,18 +33,15 @@ public static UserUtil getSingleInstance() {
...
+   * @throws RuntimeException Unable to post to DB.
    */
-  public void createAccount(String user, String pw)
-      throws IllegalArgumentException, NotImplementedException {
-    /*
-    TODO: Use Firebase API to fetch user data. \
-     Upload it to DB and return to welcome screen for login
-     */
-    throw new NotImplementedException();
+  public void postAccount(User user) throws RuntimeException {
```
Comment (danielpgp1012) [inline app/src/main/java/ch/epfl/favo/user/UserUtil.java:39]: I think in the future these void functions should return a Task that can be checked for completion and used to notify the UI

File: app/src/test/java/ch/epfl/favo/user/UserUnitTests.java [hunk 2]
```diff
@@ -102,17 +103,19 @@ public void userCanAcceptOrRequestAtMostOnce() {
...
-          @Override
-          public void run() throws Throwable {
-            UserUtil.getSingleInstance().createAccount(username, pw);
-          }
-        });
+  public void userSuccessfullyPostsToDB() {
+    CollectionWrapper mock = Mockito.mock(CollectionWrapper.class);
+    Mockito.doNothing().when(mock).addDocument(any(User.class));
+
+    String name = TestConstants.NAME;
```
Comment (danielpgp1012) [inline app/src/test/java/ch/epfl/favo/user/UserUnitTests.java:110]: We can extract these into the FakeItemFactory. (Next sprint)

File: app/src/main/java/ch/epfl/favo/user/UserUtil.java (hunk 1 above)
Comment (muneebazam) [reply app/src/main/java/ch/epfl/favo/user/UserUtil.java:39]: Agreed, I need more help with tasks/futures so ill work with alex on it this next sprint

File: app/src/test/java/ch/epfl/favo/user/UserUnitTests.java (hunk 2 above)
Comment (muneebazam) [reply app/src/test/java/ch/epfl/favo/user/UserUnitTests.java:110]: sounds good

**Pull request merged**